apache/superset
# revert(#17570): fix dashboard editing for dashboards with non-ascii characters

### SUMMARY

The bug: When a user saved a dashboard that contained a chart with non-ascii characters, the json_metadata became malformed, causing the dashboard to fail to load. The root cause is in #17570, which added `positions` into the json_metadata blob. Something went wrong in the json parsing which causes json_metadata to be truncated if the positions json contains emojis or other non-standard characters.

The root cause PR is pretty large, so we had trouble figuring out where this goes wrong. I suspect the fix forward is some combination of
* Fix json parsing to handle non-ascii characters correctly
* Validate that json_metadata is well-formed before writing it
* Don't write the `positions` json, since it's already contained by a different column

Opening this revert PR as a starting point. This PR reverts #17570 and the accompanying #17707.


### TESTING INSTRUCTIONS

to reproduce bug:
* make dashboard
* add chart with emoji in title
* save dashboard
* see error after save
* refresh and see error on load

to verify this fix:
* make dashboard
* add chart with emoji in title
* save dashboard
* verify dashboard saved
* refresh and view dashboard

### ADDITIONAL INFORMATION



@geido @etr2460 @graceguo-supercat @rusackas
